ui/StatusQ/CMakeLists.txt

```diff
@@ -2,29 +2,163 @@ cmake_minimum_required(VERSION 3.19)
+    #     - no resources are compiled, it's expected to use QML/JS sources
+    #     - the plugin is compiled in src directory
+    #
+    #   STATUSQ_SHADOW_BUILD mode is created for sandbox/storybook hot reloading
+    #   without copying all qml files to build directory.
+    #
+    #   It's expected for the app to add ${STATUSQ_MODULE_IMPORT_PATH} to
+    #   QQmlApplicationEngine::addImportPath.
+    #
+
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(STATUSQ_SHADOW_BUILD OFF)
+    else()
+        set(STATUSQ_SHADOW_BUILD ON)
+    endif()
+endif()
+
+set(STATUSQ_SOURCE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/src/StatusQ")
+
+if (${STATUSQ_SHADOW_BUILD})
+    set(STATUSQ_MODULE_PATH "${CMAKE_BINARY_DIR}/bin/StatusQ")
+    set(STATUSQ_MODULE_IMPORT_PATH ":/")
+else()
+    set(STATUSQ_MODULE_PATH "${STATUSQ_SOURCE_PATH}")
+    set(STATUSQ_MODULE_IMPORT_PATH "${CMAKE_CURRENT_SOURCE_DIR}/src")
+endif()
+
+set(STATUSQ_MODULE_PATH "${STATUSQ_MODULE_PATH}" PARENT_SCOPE)
+set(STATUSQ_MODULE_IMPORT_PATH "${STATUSQ_MODULE_IMPORT_PATH}" PARENT_SCOPE)
+
+if (${STATUSQ_SHADOW_BUILD})
+    qt5_add_big_resources(STATUSQ_QRC_COMPILED
+            src/statusq.qrc
+            src/assets.qrc)
+endif()
+
+add_library(StatusQ SHARED
+        ${STATUSQ_QRC_COMPILED}
+        src/plugin.cpp
+        include/StatusQ/QClipboardProxy.h
+        include/StatusQ/modelutilsinternal.h
+        include/StatusQ/rxvalidator.h
+        include/StatusQ/statussyntaxhighlighter.h
+        include/StatusQ/statuswindow.h
+        src/QClipboardProxy.cpp
+        src/modelutilsinternal.cpp
+        src/rxvalidator.cpp
+        src/statussyntaxhighlighter.cpp
+        src/statuswindow.cpp
+        )
+
+set_target_properties(StatusQ PROPERTIES 
+    ADDITIONAL_CLEAN_FILES bin/StatusQ/qmldir
+)
+
+if (APPLE)
+    find_library(AppKit AppKit)
+    find_library(Foundation Foundation)
+    target_link_libraries(${PROJECT_NAME} PRIVATE ${AppKit} ${Foundation})
+    target_sources(StatusQ PRIVATE src/statuswindow_osx.mm)
+else ()
+    target_sources(StatusQ PRIVATE src/statuswindow_other.cpp)
+endif ()
+
+set_target_properties(StatusQ PROPERTIES
+        RUNTIME_OUTPUT_DIRECTORY ${STATUSQ_MODULE_PATH}
+        LIBRARY_OUTPUT_DIRECTORY ${STATUSQ_MODULE_PATH}
+        )
+
+if (${STATUSQ_SHADOW_BUILD})
+    set(COPY_QMLDIR_COMMAND
+            ${CMAKE_COMMAND} -E copy
+            ${STATUSQ_SOURCE_PATH}/qmldir
+            ${STATUSQ_MODULE_PATH}/qmldir)
+    add_custom_command(
+            TARGET StatusQ
+            POST_BUILD
+            COMMAND ${COPY_QMLDIR_COMMAND}
+            COMMENT "Copying qmldir to binary directory")
+endif ()
+
+target_link_libraries(StatusQ PRIVATE
+        Qt5::Core
+        Qt5::Qml
+        Qt5::Gui
+        Qt5::Quick
+        Qt5::QuickControls2
+        SortFilterProxyModel
+        qzxing
+        )
+
+target_include_directories(StatusQ PRIVATE include)
+
+if (UNIX AND NOT APPLE)
+    target_compile_options(StatusQ PRIVATE "-fPIC")
+endif()
```

Comment:
Could you please elaborate on that? Is it needed even when `CMAKE_POSITION_INDEPENDENT_CODE` is set already?

Reply:
Ooops, this shouldn't be here. Removed!